### PR TITLE
Delombokify dough-data

### PR DESCRIPTION
**Pull request by md5sha256**
Dough data should now be compile ready

### CODE REVIEW

File: dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/OptionalMap.java [hunk 1]
```diff
@@ -184,4 +181,8 @@ public V merge(K key, V value, BiFunction<? super V, ? super V, ? extends V> fun
         return StreamSupport.stream(spliterator(), false);
     }
 
+    public Map<K, V> getInternalMap() {
```
Comment (TheBusyBiscuit) [inline dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/OptionalMap.java:184]:
```suggestion
    public @Nonnull Map<K, V> getInternalMap() {
```

File: dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/OptionalPair.java [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
 
-import lombok.Data;
-
-@Data
 public class OptionalPair<P, S> {
 
-    private Optional<P> firstValue = Optional.empty();
-    private Optional<S> secondValue = Optional.empty();
+    private Optional<P> firstValue;
+    private Optional<S> secondValue;
```
Comment (TheBusyBiscuit) [inline dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/OptionalPair.java:10]: Shouldn't these be `final`?

File: dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/WeightedNode.java [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
+        return this.object;
+    }
+
+    public void setWeight(float weight) {
+        this.weight = weight;
+    }
+
+    public void setObject(T object) {
+        this.object = object;
+    }
```
Comment (TheBusyBiscuit) [inline dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/WeightedNode.java:28]: T should always be nonnull

File: dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/Pair.java [hunk 4]
```diff
@@ -23,4 +21,45 @@ public Pair(OptionalPair<P, S> pair) {
...
+        return this.secondValue;
+    }
+
+    public void setFirstValue(P firstValue) {
+        this.firstValue = firstValue;
+    }
+
+    public void setSecondValue(S secondValue) {
+        this.secondValue = secondValue;
+    }
```
Comment (TheBusyBiscuit) [inline dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/Pair.java:38]: P and S are nullable, right?

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/OptionalPair.java (hunk 2 above)
Comment (md5sha256) [reply dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/OptionalPair.java:10]: Theres a setter for them.

File: dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/Pair.java (hunk 4 above)
Comment (md5sha256) [reply dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/Pair.java:38]: They have undefined nullability. Is there a specific annotation I should use or is `@Nullable` fine?

File: dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/OptionalPair.java (hunk 2 above)
Comment (TheBusyBiscuit) [reply dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/OptionalPair.java:12]: Alright then

File: dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/Pair.java (hunk 4 above)
Comment (TheBusyBiscuit) [reply dough-data/src/main/java/io/github/thebusybiscuit/dough/collections/Pair.java:40]: These should definitely be nullable though, see line 23 / 21:
` this(pair.getFirstValue().orElse(null), pair.getSecondValue().orElse(null));`
The value `null` is explicitly injected here. I would count this as definitely nullable.

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**